### PR TITLE
Video categories and category filtering.

### PR DESCRIPTION
**Pull request by DarryQueen**
Added video categories and implemented filtering in the search page.

Reviewer: @nalnaji @jonathanshum

### CODE REVIEW

File: app/models/video.rb [hunk 1]
```diff
@@ -4,6 +4,7 @@ class Video < ActiveRecord::Base
 
   attr_accessible :description, :title, :youtube_id, :starred, :duration
   acts_as_taggable
+  acts_as_taggable_on :category
```
Comment (nalnaji) [inline app/models/video.rb]: Why didn't we make a table that represents categories? What are the benefits of acts_as_taggable in terms of speed and efficiency? I feel like adding tags is less intuitive and more difficult than adding relations.

Comment (DarryQueen) [inline app/models/video.rb]: Simplicity. Categories don't have much value as a model, and outsourcing most of the database stuff is cleaner.

File: app/views/videos/index/_videos_table.html.erb [hunk 2]
```diff
@@ -2,7 +2,7 @@
   <thead>
     <tr>
       <th>Title</th>
-      <th>Category</th>
+      <th>Subject</th>
```
Comment (nalnaji) [inline app/views/videos/index/_videos_table.html.erb]: We should be consistent in the way we reference these. If it's "Subject" to the user, it should be "Subject" to the developers too.

File: app/models/video.rb (hunk 1 above)
Comment (nalnaji) [inline app/models/video.rb]: I take back what I said, CSV make it simple.

File: app/views/videos/index/_videos_table.html.erb (hunk 2 above)
Comment (DarryQueen) [inline app/views/videos/index/_videos_table.html.erb]: Changing to subject.

**Comment by DarryQueen** [comment]
All instances of category now changed to subject.

File: app/views/videos/index.html.erb [hunk 3]
```diff
@@ -1,25 +1,27 @@
...
-    <div class="col-lg-8 col-md-6 no-padding">
-      <%= autocomplete_field_tag :search, nil, autocomplete_tag_name_videos_path, {
-        :id => 'search',
-        :class => 'form-control video-search',
-        :placeholder => 'Search by category or name.',
-        :'data-delimiter' => ' ',
-        :tabindex => '-1',
+    <div class="col-lg-6 col-md-5 no-padding">
+      <%= text_field_tag :search, nil, {
+        :class => 'form-control', :placeholder => 'Search by title or category.', :autocomplete => 'off'
```
Comment (nalnaji) [inline app/views/videos/index.html.erb]: Lol.

Comment (DarryQueen) [inline app/views/videos/index.html.erb]: That placeholder has been shitty for a while. We're planning to remove search by subject once we finish case-insensitive search. We'll update placeholder along with it.

**Comment by nalnaji** [comment]
Cool, lets do it. Merge

**Pull request merged**